numpy/core/include/numpy/npy_common.h

```diff
@@ -61,6 +61,13 @@
 #define NPY_UNLIKELY(x) (x)
 #endif
 
+#ifdef HAVE___BUILTIN_PREFETCH
+/* unlike _mm_prefetch also works on non-x86 */
+#define NPY_PREFETCH(x, rw, loc) __builtin_prefetch((x), (rw), (loc))
+#else
+#define NPY_PREFETCH(x, rw,loc)
```

Comment:
its just ugly and relies in the compiler doing constant propagation properly as the hint must be a constant
though if we skip the _MM_HINT_ET[01] its somewhat reasonable